test/12_conditional_commit.ts

```diff
@@ -38,379 +63,507 @@ let BosonRouter_Factory: ContractFactory;
+            );
+
+            const promiseSeller = await contractVoucherKernel.getSupplyHolder(
+              tokenSupplyKey
+            );
+
+            assert.strictEqual(
+              promiseSeller,
+              users.seller.address,
+              'Seller incorrect'
+            );
+
+            const promiseOrderData = await contractVoucherKernel.getOrderCosts(
+              tokenSupplyKey
+            );
+            assert.isTrue(
+              promiseOrderData[constants.PROMISE_ORDER_FIELDS.price].eq(
+                BN(constants.PROMISE_PRICE1)
+              ),
+              'Promise produt price mismatch'
+            );
+            assert.isTrue(
+              promiseOrderData[constants.PROMISE_ORDER_FIELDS.depositSe].eq(
+                BN(constants.PROMISE_DEPOSITSE1)
+              ),
+              'Promise seller deposit mismatch'
+            );
+            assert.isTrue(
+              promiseOrderData[constants.PROMISE_ORDER_FIELDS.depositBu].eq(
+                BN(constants.PROMISE_DEPOSITBU1)
+              ),
+              'Promise buyer deposit mismatch'
+            );
+
+            const tokenNonce = await contractVoucherKernel.getTokenNonce(
+              users.seller.address
+            );
+            assert.isTrue(
+              tokenNonce.eq(constants.ONE),
+              'Voucher kernel nonce mismatch'
+            );
+
+            assert.equal(
+              promiseId,
+              await contractVoucherKernel.getPromiseIdFromSupplyId(
+                tokenSupplyKey
+              ),
+              'PromisId mismatch'
+            );
+
+            // Check VoucherSets state
+            const sellerVoucherSetsBalance = (
+              await contractVoucherSets.functions[fnSignatures.balanceOf1155](
+                users.seller.address,
+                tokenSupplyKey
+              )
+            )[0];
+
+            assert.isTrue(
+              sellerVoucherSetsBalance.eq(constants.QTY_10),
+              'VoucherSets seller balance mismatch'
+            );
+          });
+
+          describe('Flow with automatic gate.registerVoucherSetId', () => {
+            it('Should be able to create Voucher with gate address and non empty nft token id', async () => {
+              const txValue = BN(constants.PROMISE_DEPOSITSE1).mul(
+                BN(constants.QTY_10)
+              );
+              expect(
+                await contractBosonRouter
+                  .connect(users.seller.signer)
+                  .requestCreateOrderETHETHConditional(
+                    [
+                      constants.PROMISE_VALID_FROM,
+                      constants.PROMISE_VALID_TO,
+                      constants.PROMISE_PRICE1,
+                      constants.PROMISE_DEPOSITSE1,
+                      constants.PROMISE_DEPOSITBU1,
+                      constants.QTY_10,
+                    ],
+                    contractGate.address,
+                    constants.NFT_TOKEN_ID,
```

Comment:
It would be nice to rename `NFT_TOKEN_ID` to `CONDITIONAL_TOKEN_ID` to be in line with other changes (applies in other places as well)

Reply:
Same in the `it `

Reply:
Good call. I replaced `NFT_TOKEN_ID` with `CONDITIONAL_TOKEN_ID` throughout the codebase which included `constants.ts`, `10_erc155_non_transferrable.ts`, `11_gate.ts`, and `12_conditional_commit.ts`